### PR TITLE
Set assert size constant

### PR DESCRIPTION
**Pull request by DJKessler**
| | |
|:---|:---|
|**_Related Issue(s)_**| #2891 |
|**_Has Unit Tests (y/n)_**| y |
|**_Documentation Included (y/n)_**| y |

---
Updated assert size constant to use an FPP constant.

- Added FPP constant `FwAssertTextSize`
- The `FW_ASSERT_TEXT_SIZE` is now defined in terms of the FPP constant `FwAssertTextSize`
- Replaced `FW_ASSERT_DFL_MSG_LEN` with `FW_ASSERT_TEXT_SIZE`
- Removed `FW_ASSERT_DFL_MSG_LEN`
- Added `config` to module dependencies to correct build errors
- Fixed a compile error caused by _unrelated_ code

### CODE REVIEW

File: Os/Stub/ConditionVariable.hpp
Comment (LeStarch) [inline Os/Stub/ConditionVariable.hpp:42]: Note: using `virtual` on a line marked `override` is redundant.  Not a blocking issue.

**Review by LeStarch — APPROVED** [review]

**Comment by LeStarch** [comment]
@thomas-bc another set of eyes before I merge?

**Review by thomas-bc — APPROVED** [review]

**Comment by thomas-bc** [comment]
LGTM!

**Pull request merged**
